tests/chainer_tests/initializer_tests/test_normal.py

```diff
@@ -111,27 +124,10 @@ def check_initializer_statistics(self, xp, n):
 
     @testing.with_requires('scipy')
     @condition.retry(3)
-    def test_initializer_statistics_cpu(self):
-        self.check_initializer_statistics(numpy, 100)
-
-    @attr.gpu
-    @testing.with_requires('scipy')
-    @condition.retry(3)
-    def test_initializer_statistics_gpu(self):
-        self.check_initializer_statistics(cuda.cupy, 100)
+    def test_initializer_statistics(self, backend_config):
+        self.check_initializer_statistics(backend_config.xp, 100)
```

Comment:
You shoud pass `backend_config` instead of `xp`.
`xp.empty(...)` ignores the CUDA device ID.